packages/js/src/utils/generators.test.ts

```diff
@@ -0,0 +1,76 @@
+import { toUint8Array } from 'js-base64';
+
+import { generateCodeChallenge, generateCodeVerifier, generateState } from './generators';
+
+const isUrlSafe = (input: string) => /^[\w-]*$/.test(input);
```

Comment:
we can add this into `essentials`

Reply:
no hurries, you can add a linear issue and do it later

Reply:
use `UrlSafeBase64` in #128 , you can take a look

Reply:
ok

Reply:
Added issue LOG-1378 and will do it later